src/timely-util/src/operator.rs

```diff
@@ -777,7 +794,7 @@ pub fn consolidate_pact<Ba, P, G>(
     stream: &StreamCore<G, Ba::Input>,
     pact: P,
     name: &str,
-) -> StreamCore<G, Ba::Output>
+) -> StreamCore<G, Rc<Vec<Ba::Output>>>
```

Comment:
The purpose of the `Rc` isn't clear to me. Is it a safe-guard to avoid accidental expensive cloning of `Vec`s when the output stream is later forked?

Reply:
It is so that a chain of updates is inseparable, no matter what channel it traverses.

Reply:
The concern being that Timely might otherwise try to pull the `Vec` apart while passing it through channels? Would be great to have a comment explaining this!

Reply:
I changed the PR such that `consolidate_pact` produces a `Stream<G, Vec<Ba::Output>>`, which means that the _elements_ we're transferring are `Vec<_>`. This gives us the certainty that Timely will not look into the chains.